File: src/leetcode/2026/01/Solution561.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution561 {
public:
    int arrayPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int sum = 0;
        for (size_t i = 0; i < nums.size(); i += 2) {
            sum += nums[i];
        }
        return sum;
    }
};

int main() {
    Solution561 solution;

    vector<int> nums1 = {1, 4, 3, 2};
    int result1 = solution.arrayPairSum(nums1);
    cout << "Maximized Sum of Min Pairs: " << result1 << endl; // Output: 4

    vector<int> nums2 = {6, 2, 6, 5, 1, 2};
    int result2 = solution.arrayPairSum(nums2);
    cout << "Maximized Sum of Min Pairs: " << result2 << endl; // Output: 9

    return 0;
}

File: src/leetcode/2026/01/Solution561.java
import java.util.Arrays;

public class Solution561 {
    public int arrayPairSum(int[] nums) {
        Arrays.sort(nums);
        int sum = 0;
        for (int i = 0; i < nums.length; i += 2) {
            sum += nums[i];
        }
        return sum;
    }

    public static void main(String[] args) {
        Solution561 solution = new Solution561();

        int[] input1 = { 1, 4, 3, 2 };
        int output1 = solution.arrayPairSum(input1);
        System.out.println(output1); // Output: 4

        int[] input2 = { 6, 2, 6, 5, 1, 2 };
        int output2 = solution.arrayPairSum(input2);
        System.out.println(output2); // Output: 9
    }
}
